speechbrain/utils/seed.py

```diff
@@ -21,7 +20,12 @@
 def seed_everything(
     seed: int = 0, verbose: bool = True, deterministic: bool = False
 ) -> int:
-    r"""Function that sets the seed for pseudo-random number generators in: torch, numpy, and Python's random module.
+    r"""Function that sets the seed for pseudo-random number generators in: torch, numpy, and Python's random module. Important note on DDP: all DDP
+    process have the same seed. This is important to ensure that parameters
+    without a require_grad set to True are the same across processes. This
+    must be taken into account if one wants to build a custom data sampler as
+    the processes would pick the same samples... SpeechBrain takes care of that
+    internally.
```

Comment:
Have we verified that the sampler actually does work correctly when they're all the same seed? What were the original reasons for setting separate seeds? Also, do we need to provide a mechanism to access a process-specific seed?

Reply:
The reason was that we thought it was not the case, but @Gastron  did the distributed sampler well :p